omnipaxos_core/tests/reconnect_test.rs

```diff
@@ -0,0 +1,260 @@
+        .find(|x| *x != leader_id)
+        .expect("No followers found!");
+    let follower = sys.nodes.get(&follower_id).unwrap();
+
+    // Decide entries during omission period
+    leader.on_definition(|comp| {
+        comp.set_connection(follower_id, false);
+    });
+    make_proposals(&sys, &cfg, leader_id, 5, 10);
+
+    // Decide entries after omission period so follower finds seq break
+    leader.on_definition(|comp| {
+        comp.set_connection(follower_id, true);
+    });
+    make_proposals(&sys, &cfg, leader_id, 10, 15);
+
+    // Wait for Re-sync with leader to finish
+    thread::sleep(SLEEP_TIMEOUT);
+
+    // Verify log
+    let expected_log = (1..15).into_iter().map(|v| Value(v)).collect();
+    let followers_log: Vec<LogEntry<Value, LatestValue>> = follower.on_definition(|comp| {
+        comp.paxos
+            .read_decided_suffix(0)
+            .expect("Cannot read decided log entry")
+    });
+    verify_log_after_recovery(followers_log, expected_log, 14);
+
+    // Shutdown system
+    println!("Passed reconnect_to_leader_test!");
+    let kompact_system =
+        std::mem::take(&mut sys.kompact_system).expect("No KompactSystem in memory");
+    match kompact_system.shutdown() {
+        Ok(_) => {}
+        Err(e) => panic!("Error on kompact shutdown: {}", e),
+    };
+}
+
+/// Return the elected leader. If there is no leader yet then
+/// wait until a leader is elected in the allocated time.
+fn get_elected_leader(sys: &TestSystem, wait_timeout: Duration) -> u64 {
+    let first_node = sys.nodes.get(&1).expect("No BLE component found");
+
+    first_node.on_definition(|x| {
+        let leader_pid = x.paxos.get_current_leader();
+        leader_pid.unwrap_or_else(|| {
+            // Leader is not elected yet
+            let (kprom, kfuture) = promise::<Ballot>();
+            x.election_futures.push(Ask::new(kprom, ()));
+
+            let ballot = kfuture
+                .wait_timeout(wait_timeout)
+                .expect("No leader has been elected in the allocated time!");
+            ballot.pid
+        })
+    })
+}
+
+/// Use node `proposer` to propose values in the range `from..to`, then waits for the proposals
+/// to be decided.
+fn make_proposals(sys: &TestSystem, cfg: &TestConfig, proposer: u64, from: u64, to: u64) {
+    let proposer = sys
+        .nodes
+        .get(&proposer)
+        .expect("No SequencePaxos component found");
+
+    let mut proposal_futures = vec![];
+    for i in from..to {
+        let (kprom, kfuture) = promise::<Value>();
+        proposer.on_definition(|x| {
+            x.paxos.append(Value(i)).expect("Failed to append");
+            x.decided_futures.push(Ask::new(kprom, ()));
+        });
+        proposal_futures.push(kfuture);
+    }
+
+    match FutureCollection::collect_with_timeout::<Vec<_>>(proposal_futures, cfg.wait_timeout) {
+        Ok(_) => {}
+        Err(e) => panic!("Error on collecting futures of decided proposals: {}", e),
+    }
+}
```

Comment:
I'm not sure if we already have something similar in `tests/util.rs`. If we have use them, else move these there so we can reuse them in other tests.